jax_galsim/position.py

```diff
@@ -0,0 +1,150 @@
+import jax.numpy as jnp
+
+import galsim as _galsim
+from jax._src.numpy.util import _wraps
+from jax.tree_util import register_pytree_node_class
+
+
+@_wraps(_galsim.Position)
+class Position(object):
+    def __init__(self, *args, **kwargs):
+        if len(kwargs) == 0:
+            if len(args) == 2:
+                self.x, self.y = args
+            elif len(args) == 0:
+                self.x = self.y = 0
+            elif len(args) == 1:
+                if isinstance(args[0], (Position,)):
+                    self.x = args[0].x
+                    self.y = args[0].y
+                else:
+                    try:
+                        self.x, self.y = args[0]
+                    except (TypeError, ValueError):
+                        raise TypeError(
+                            "Single argument to %s must be either a Position "
+                            "or a tuple." % self.__class__
+                        )
+            else:
+                raise TypeError(
+                    "%s takes at most 2 arguments (%d given)"
+                    % (self.__class__, len(args))
+                )
+        elif len(args) != 0:
+            raise TypeError(
+                "%s takes x and y as either named or unnamed arguments (given %s, %s)"
+                % (self.__class__, args, kwargs)
+            )
+        else:
+            try:
+                self.x = kwargs.pop("x")
+                self.y = kwargs.pop("y")
+            except KeyError:
+                raise TypeError(
+                    "Keyword arguments x,y are required for %s" % self.__class__
+                )
+            if kwargs:
+                raise TypeError("Got unexpected keyword arguments %s" % kwargs.keys())
+
+        # Make sure the inputs are indeed jax numpy values
+        self.x = jnp.asarray(self.x)
+        self.y = jnp.asarray(self.y)
```

Comment:
Here I noticed that `jnp.asarray` truncates the input to `float32`, so this might be an issue if we work with `float64` at some point.

Reply:
Does this happen even if you set the float64 flag for JAX at the initialization?

Reply:
note that I have included the following in the conftest.py:
```
# Define the accuracy for running the tests
from jax.config import config

config.update("jax_enable_x64", True)
```

Reply:
Ah it works with this config :-) I was doing my own tests and forgot this